sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ForLoopIncrementSignCheck.cs

```diff
@@ -34,25 +34,206 @@ namespace SonarAnalyzer.Rules.CSharp
+            Substraction
+        }
+
+        private enum Condition
+        {
+            None,
+            Less,
+            Greater
+        }
+
+        private readonly Dictionary<SyntaxKind, ArithmeticOperation> incrementorToOperation = new Dictionary<SyntaxKind, ArithmeticOperation>
+        {
+            { SyntaxKind.PreDecrementExpression, ArithmeticOperation.Substraction },
+            { SyntaxKind.PreIncrementExpression, ArithmeticOperation.Addition },
+            { SyntaxKind.PostDecrementExpression, ArithmeticOperation.Substraction },
+            { SyntaxKind.PostIncrementExpression, ArithmeticOperation.Addition },
+            { SyntaxKind.SubtractAssignmentExpression, ArithmeticOperation.Substraction },
+            { SyntaxKind.AddAssignmentExpression, ArithmeticOperation.Addition },
+            { SyntaxKind.AddExpression, ArithmeticOperation.Addition },
+            { SyntaxKind.SubtractExpression, ArithmeticOperation.Substraction }
+        };
+
         protected override void Initialize(SonarAnalysisContext context)
         {
             context.RegisterSyntaxNodeActionInNonGenerated(c =>
                 {
-                    var node = c.Node;
-                    if (true)
+                    var forNode = (ForStatementSyntax)c.Node;
+
+                    var conditionSyntax = forNode.Condition;
+                    if (conditionSyntax == null || !(conditionSyntax is BinaryExpressionSyntax binaryCondition))
                     {
-                        c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, node.GetLocation()));
+                        return;
+                    }
+
+                    if (forNode.Incrementors.Count != 1)
+                    {
+                        return;
+                    }
+
+                    var incrementor = forNode.Incrementors[0];
+
+                    var incrementorData = GetIncrementData(incrementor);
+                    if (incrementorData.Operation == ArithmeticOperation.None)
+                    {
+                        return;
+                    }
+
+                    var condition = GetCondition(binaryCondition, incrementorData.IdentifierName);
+                    if (condition == Condition.None)
+                    {
+                        return;
+                    }
+
+                    if (incrementorData.Operation == ArithmeticOperation.Addition &&
+                        condition == Condition.Greater)
+                    {
+                        c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, forNode.Condition.GetLocation(), incrementorData.IdentifierName, "inc"));
+                    }
+                    else if (incrementorData.Operation == ArithmeticOperation.Substraction &&
+                             condition == Condition.Less)
+                    {
+                        c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, forNode.Condition.GetLocation(), incrementorData.IdentifierName, "dec"));
```

Comment:
in [sonar-php](https://github.com/SonarSource/sonar-php/blob/3875b7574ba3be9f2fb6ce21be815547b6f2642a/php-checks/src/main/java/org/sonar/php/checks/ForLoopIncrementSignCheck.java) there is a secondary location labeled "stop condition" - by adding the primary location as the `indexer` and the secondary location as the `condition`, the message will make more sense

Reply:
Done